dashboard/app/models/level.rb

```diff
@@ -239,6 +239,14 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  # Find the first script this level belongs to that is a HOC script.
+  # Returns nil if this level does not belong to any HOC scripts.
+  def find_hoc_script
+    hoc_script_level = self.script_levels.find{|l| l.script.hoc?}
```

Comment:
Can you add testing for the level cache?

On Wed, Nov 25, 2015 at 11:38 AM, Andrew Oberhardt <notifications@github.com
> wrote:

> In dashboard/app/models/level.rb
> <https://github.com/code-dot-org/code-dot-org/pull/5617#discussion_r45910673>
> :
>
> > @@ -239,6 +239,14 @@ def strip_name
> >      self.name = name.to_s.strip unless name.nil?
> >    end
> >
> > +  # Find the first script this level belongs to that is a HOC script.
> > +  # Returns nil if this level does not belong to any HOC scripts.
> > +  def find_hoc_script
> > +    hoc_script_level = self.script_levels.find{|l| l.script.hoc?}
>
> I went ahead and added a level cache in the script model with the other
> caches. @laurelfan <https://github.com/laurelfan> please take a look.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/5617/files#r45910673>.
>